### PR TITLE
feat(compiler-cli): Adapt the compiler to use standalone by default. 

### PR DESCRIPTION
**Pull request by JeanMeche**
On top of #58160 & #58175

This PR flips the default for standalone from `false` to `true` for components, directives and pipes. 
This PR also includes the necessary changes for the impacted tests as well as other targets that dependens on them (adev, benchmarks etc).

### CODE REVIEW

File: packages/core/src/render3/jit/standalone-default-value.ts [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+/**
+ * A constant defining the default value for the standalone attribute in Directive and Pipes decorators.
+ * Extracted to a separate file to facilitate G3 patches.
+ */
+export const NG_STANDALONE_DEFAULT_VALUE = true;
```
Comment (AndrewKushnir) [inline packages/core/src/render3/jit/standalone-default-value.ts:13]: I think it'd be great to land this change as a refactoring first, without changing the default. After that, we can land another PR that would flip the flag. We'll need those 2 steps to reduce risks of landing this change in g3.

Comment (JeanMeche) [reply packages/core/src/render3/jit/standalone-default-value.ts:13]: Great suggestion, It'll split the change even more.

Comment (JeanMeche) [reply packages/core/src/render3/jit/standalone-default-value.ts]: #58175 will be the intermediate PR.

File: packages/compiler-cli/linker/src/file_linker/partial_linkers/util.ts [hunk 2]
```diff
@@ -113,3 +114,11 @@ export function extractForwardRef<TExpression>(
     ForwardRefHandling.Unwrapped,
   );
 }
+
+const STANDALONE_IS_DEFAULT_RANGE = new semver.Range('>= 19.0.0', {
```
Comment (alxhub) [inline packages/compiler-cli/linker/src/file_linker/partial_linkers/util.ts:118]: We should make sure that `0.0.0-PLACEHOLDER` is also considered to pass.

**Review by alxhub — APPROVED** [review]

**Review by devversion — APPROVED** [review]

**Comment by JeanMeche** [comment]
This change has currently an issue in our compliance tests.
Tests that rely on explicit `standalone: false` aren't compiled correctly by the linked target.

**Review by jelbourn — APPROVED** [review]
LGTM

Reviewed-for: global-approvers

**Comment by jelbourn** [comment]
Caretaker note: presubmit failure looks like a flake to me; I ran a TGP for these changes yesterday and it was green

**Comment by devversion** [comment]
This PR was merged into the repository by commit 9ab663e7a29da63df03f453678ae9ed441c67a5d.

The changes were merged into the following branches: main
